src/analysis/normed_space/weak_dual.lean

```diff
@@ -182,3 +190,255 @@ begin
 end
 
 end polar_sets_in_weak_dual
+
+section embedding_to_Pi
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+variables {E : Type*} [normed_group E] [normed_space 𝕜 E]
+
+open metric set normed_space
+
+/-- The (weak) dual `weak_dual 𝕜 E` of a normed space `E` consists of bounded linear
+functionals `E → 𝕜`. Such functionals can be interpreted as elements of the Cartesian
+product `E → 𝕜` via the function `weak_dual.to_Pi : weak_dual 𝕜 E → (E → 𝕜)`. -/
+def _root_.weak_dual.to_Pi (𝕜 : Type*) [nondiscrete_normed_field 𝕜]
+  (E : Type*) [topological_space E] [add_comm_group E] [module 𝕜 E] (x' : weak_dual 𝕜 E) :=
+x'.to_fun
+
+/-- For any `f : Π (j : ι), X j` in a product of metric spaces `X j`, a set of the
+form `{g | dist (f i) (g i) < ε}` with `ε > 0` and `i : ι` is a neighborhood of `f`. -/
+lemma _root_.mem_nhds_Pi' {ι : Type*} {X : ι → Type*} [∀ (i : ι), metric_space (X i)]
+  (f : (Π (j : ι), X j)) (i : ι) {ε : ℝ} (ε_pos : 0 < ε) :
+  {g : (Π (j : ι), X j) | dist (f i) (g i) < ε} ∈ 𝓝 f :=
+begin
+  have nhd : {z : X i | dist (f i) z < ε} ∈ 𝓝 (f i),
+  by { simp_rw dist_comm, exact ball_mem_nhds (f i) ε_pos, },
+  have key := set_pi_mem_nhds (finite_singleton i)
+    (by { intros j hj, rwa mem_singleton_iff.mp hj, } :
+    ∀ j ∈ {i}, {z : X j | dist (f j) z < ε} ∈ 𝓝 (f j)),
+  simp only [singleton_pi, preimage_set_of_eq, function.eval_apply] at key,
+  exact key,
+end
+
+/-- For any `f : Π (j : ι), K j` in a product of normed groups `K j`, a set of the
+form `{g | dist (f i) (g i) < ε}` with `ε > 0` and `i : ι` is a neighborhood of `f`. -/
+lemma _root_.mem_nhds_Pi_normed_field {ι : Type*} {K : ι → Type*} [∀ (i : ι), normed_group (K i)]
+  (f : (Π j, K j)) (i : ι) {ε : ℝ} (ε_pos : 0 < ε) :
+  {g : (Π j, K j) | ∥ f i - g i ∥ < ε} ∈ 𝓝 f :=
+begin
+  have key := _root_.mem_nhds_Pi' f i ε_pos,
+  rwa (_ : {g : (Π j, K j) | ∥f i - g i∥ < ε} = {g : (Π j, K j) | dist (f i) (g i) < ε}),
+  simp only [dist_eq_norm],
+end
+
+/-- The function `weak_dual.to_Pi : weak_dual 𝕜 E → (E → 𝕜)` is an embedding. -/
+lemma weak_dual.to_Pi_embedding (𝕜 : Type*) [nondiscrete_normed_field 𝕜]
+  (E : Type*) [topological_space E] [add_comm_group E] [module 𝕜 E] :
+  embedding (weak_dual.to_Pi 𝕜 E) :=
+{ induced := eq_of_nhds_eq_nhds (congr_fun rfl),
+  inj := by { intros φ₁ φ₂ h, ext z, exact congr_fun h z, }, }
```

Comment:
Does `induced := rfl`, `inj := continuous_linear_map.coe_fn_injective` work?

Reply:
It does! Thank you! Changed.